Content.Server/_CorvaxGoob/Deathmatch-CS/Observer.cs

```diff
@@ -0,0 +1,45 @@
+using Content.Server.GameTicking;
+using Content.Server.Ghost.Roles.Components;
+using Content.Shared.GameTicking.Components;
+using Content.Shared.Mind.Components;
+using Content.Shared.Mobs.Systems;
+
+namespace Content.Server._CorvaxGoob.Deathmatch_CS;
+
+public sealed class CSObserver : EntitySystem
+{
+    [Dependency] private readonly GameTicker _gameTicker = default!;
+    [Dependency] private readonly CSRuleSystem _csRuleSystem = default!;
+    [Dependency] private readonly MobStateSystem _mobStateSystem = default!;
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<FighterComponent, MindAddedMessage>(OnMindAdded);
+    }
+
+    private void OnMindAdded(EntityUid uid, FighterComponent component, MindAddedMessage args)
+    {
+        var query = EntityQueryEnumerator<CSRuleComponent, GameRuleComponent>();
+        while (query.MoveNext(out var uId, out var csRuleC, out var gRuleC))
```

Comment:
```suggestion

        while (query.MoveNext(out var uId, out var _, out var gRuleC))
```
читабельность. csRuleC вообще не используется, значит и не надо память им занимать